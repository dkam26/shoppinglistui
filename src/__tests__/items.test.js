import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import Items  from '../components/items.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import toJson,{ shallowToJson } from 'enzyme-to-json';
import LocalStorageMock from '../setupTests';
import ReactDOM from 'react-dom';
configure({ adapter: new Adapter() });
describe('items component', () => {
   
    const props={
        constructor :()=>{},
        render:()=>{},
    }
    const wrapper =  shallow(<Items />);
     it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    }); 
    it('should render correctly', () => {
        const output = shallow( <Items />);
        expect(shallowToJson(output)).toMatchSnapshot();
        });  
    it('should add item',()=>{
            wrapper.find('#addItem').simulate('click')
            expect(toJson(wrapper)).toMatchSnapshot();    
        });
    it('renders without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(<Items />, div);
        });  
  
})