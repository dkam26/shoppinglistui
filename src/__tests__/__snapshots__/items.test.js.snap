// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`items component should add item 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuItem>
            <Popup
              content="Add item"
              id="addItem"
              on="hover"
              position="top left"
              trigger={
                <Button
                  as="button"
                  icon="add"
                  onClick={[Function]}
                />
              }
            />
          </MenuItem>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Items in 
               list
            </h3>
          </MenuItem>
        </MenuMenu>
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Input
              icon="search"
              name="word"
              onChange={[Function]}
              onKeyUp={[Function]}
              placeholder="Search by name..."
              type="text"
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Button
              as="button"
              content="Back"
              icon="left chevron"
              labelPosition="left"
              onClick={[Function]}
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <p>
      No items currently
    </p>
    <_class
      options={Object {}}
    />
  </Container>
</div>
`;

exports[`items component should render correctly 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuItem>
            <Popup
              content="Add item"
              id="addItem"
              on="hover"
              position="top left"
              trigger={
                <Button
                  as="button"
                  icon="add"
                  onClick={[Function]}
                />
              }
            />
          </MenuItem>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Items in 
               list
            </h3>
          </MenuItem>
        </MenuMenu>
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Input
              icon="search"
              name="word"
              onChange={[Function]}
              onKeyUp={[Function]}
              placeholder="Search by name..."
              type="text"
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Button
              as="button"
              content="Back"
              icon="left chevron"
              labelPosition="left"
              onClick={[Function]}
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <p>
      No items currently
    </p>
    <_class
      options={Object {}}
    />
  </Container>
</div>
`;
